### PR TITLE
Added additional AMQP logging

### PR DESCRIPTION
**Pull request by abhipsaMisra**
## Checklist
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [x] This pull-request is submitted against the `master` branch.

## Description of the changes

Added additional AMQP logging

### CODE REVIEW

File: iothub/device/src/Common/Amqp/LegacyClientWebSocketTransport.cs [hunk 1]
```diff
@@ -66,6 +67,8 @@ public override void SetMonitor(ITransportMonitor usageMeter)
 
         public override bool WriteAsync(TransportAsyncCallbackArgs args)
         {
+            if (Logging.IsEnabled) Logging.Enter(this, args.Transport, $"{nameof(LegacyClientWebSocketTransport)}.{nameof(WriteAsync)}");
```
Comment (CIPop) [inline iothub/device/src/Common/Amqp/LegacyClientWebSocketTransport.cs]: Missing Logging.Exit

File: iothub/device/src/Common/FaultTolerantAmqpObject.cs [hunk 2]
```diff
@@ -23,6 +24,8 @@ public FaultTolerantAmqpObject(Func<TimeSpan, CancellationToken, Task<T> > creat
 
         public bool TryGetOpenedObject(out T openedAmqpObject)
         {
+            if (Logging.IsEnabled) Logging.Enter(this, $"{nameof(FaultTolerantAmqpObject<AmqpObject>)}.{nameof(TryGetOpenedObject)}");
```
Comment (CIPop) [inline iothub/device/src/Common/FaultTolerantAmqpObject.cs]: Missing Exit statements.

File: iothub/device/src/Common/FaultTolerantAmqpObject.cs [hunk 3]
```diff
@@ -42,6 +45,8 @@ public bool TryGetOpenedObject(out T openedAmqpObject)
 
         protected override async Task<T> OnCreateAsync(TimeSpan timeout, CancellationToken cancellationToken)
         {
+            if (Logging.IsEnabled) Logging.Enter(this, timeout, cancellationToken, $"{nameof(FaultTolerantAmqpObject<AmqpObject>)}.{nameof(OnCreateAsync)}");
```
Comment (CIPop) [inline iothub/device/src/Common/FaultTolerantAmqpObject.cs]: Missing exit

File: iothub/device/src/Common/FaultTolerantAmqpObject.cs [hunk 4]
```diff
@@ -50,6 +55,8 @@ protected override async Task<T> OnCreateAsync(TimeSpan timeout, CancellationTok
 
         protected override void OnSafeClose(T value)
         {
+            if (Logging.IsEnabled) Logging.Enter(this, value.Identifier, $"{nameof(FaultTolerantAmqpObject<AmqpObject>)}.{nameof(OnSafeClose)}");
```
Comment (CIPop) [inline iothub/device/src/Common/FaultTolerantAmqpObject.cs]: Missing exit

File: iothub/device/src/IotHubConnection.cs [hunk 5]
```diff
@@ -77,6 +77,8 @@ protected IotHubConnection(string hostName, int port, AmqpTransportSettings amqp
             ProductInfo productInfo,
             CancellationToken cancellationToken)
         {
+            if (Logging.IsEnabled) Logging.Enter(this, corrId, linkType, $"{nameof(IotHubConnection)}.{nameof(CreateSendingLinkAsync)}");
```
Comment (CIPop) [inline iothub/device/src/IotHubConnection.cs]: Missing exit statements in this file.

Comment (CIPop) [reply iothub/device/src/IotHubConnection.cs]: Here and everywhere in the file.

**Comment by CIPop** [comment]
@abhipsaMisra: your changes are breaking the NetMF build.

**Comment by abhipsaMisra** [comment]
Updated with the fix

**Pull request merged**